### PR TITLE
Streamline ADC data collection

### LINKED ISSUE #15: BLE transmit protocol might send a partial adc reading
Currently, the firmware sends up a payload that is bigger than the 200 byte threshold. Since the BLE task is fast enough, usually there isn't a build up in the buffer. Each adc reading is 15 bytes, so 14 readings get sent - `14*15 = 210 bytes`.
But if the task slows down and the buffer builds up, we might send 244 bytes, which means that a partial adc reading will be sent. `16*15 = 240 bytes`, so 16 full readings + 4 bytes of the next reading. From that point on, all packets will have partial adc reading in it.

I don't think this is very nice, so we should probably change the bytes read from the buffer to always be a multiple of the adc reading.

### PR DESCRIPTION
**Pull request by dl-dl**
General cleanup.
Explicit BLE ADC feed data format.
Fixes #15

### CODE REVIEW

File: main/main.cpp [hunk 1]
```diff
@@ -119,36 +130,17 @@ void IRAM_ATTR isrAdcDrdy() {
...
-	buffer[STATUS_SIZE + NUM_CHAN * SAMPLE_SIZE] = adcReading.crc_match;
-
-	// Only place ADC reading if the the BLE device is connected.
-	// TODO - this probably should be handled in a different way.
-	xStreamBufferSend(adcStreamBufferHandle, buffer, sizeof(buffer), 0);
-
+	BlePublAdcData toSend{.status = adcReading.status,
+	                      .data   = adcReading.data[sizeof(BlePublAdcData::data)],
+	                      .crc    = adc.isCrcOk(&adcReading)};
+	xStreamBufferSend(adcStreamBufferHandle, &toSend, sizeof(toSend), 0);
```
Comment (larryk3) [inline main/main.cpp:141]:
```suggestion
	memcpy(toSend.data, adcReading.data, sizeof(toSend.data));
	xStreamBufferSend(adcStreamBufferHandle, &toSend, sizeof(toSend), 0);
```

**Review by larryk3 — CHANGES_REQUESTED** [review]
toSend data does not get copied over properly

**Review by larryk3 — CHANGES_REQUESTED** [review]

File: main/main.cpp [hunk 2]
```diff
@@ -38,7 +38,18 @@ constexpr uint32_t CORE_APP = 1;
 // Nimble creates a GATT connection, which has some overhead.
 constexpr uint16_t BLE_PUBL_DATA_DLE         = 251;
 constexpr uint16_t BLE_PUBL_DATA_ATT_PAYLOAD = BLE_PUBL_DATA_DLE - 4 - 3;
-constexpr size_t   BLE_PUBL_DATA_TRIGGER     = 200;
+
+#pragma pack(push, 1)
+struct BlePublAdcData {
```
[the remark this answers is not in the document]
Comment (dl-dl) [reply main/main.cpp:43]: Ok. Done.

File: main/main.cpp [hunk 3]
```diff
@@ -300,7 +294,7 @@ extern "C" void app_main(void) {
 	                        &bleAdcFeedPublisherTaskHandle, CORE_BLE);
 
 	// // esp_pm_config_esp32_t pm_config = {
-	const esp_pm_config_t pmConfig = {
+	esp_pm_config_t pmConfig = {
```
[the remark this answers is not in the document]
Comment (dl-dl) [reply main/main.cpp:297]: Probably by mistake. Reinstated.

File: main/main.cpp [hunk 4]
```diff
@@ -119,36 +130,17 @@ void IRAM_ATTR isrAdcDrdy() {
...
-	// Pack CRC match as a single byte
-	buffer[STATUS_SIZE + NUM_CHAN * SAMPLE_SIZE] = adcReading.crc_match;
-
-	// Only place ADC reading if the the BLE device is connected.
-	// TODO - this probably should be handled in a different way.
-	xStreamBufferSend(adcStreamBufferHandle, buffer, sizeof(buffer), 0);
-
+	BlePublAdcData toSend{.status = adcReading.status, .data = {}, .crc = adc.isCrcOk(&adcReading)};
+	static_assert(sizeof(toSend.data) == sizeof(adcReading.data));
+	memcpy(toSend.data, adcReading.data, sizeof(toSend.data));
```
[the remark this answers is not in the document]
Comment (dl-dl) [reply main/main.cpp:140]: Let's return to it later, when we know what exactly we need to send.

File: main/main.cpp [hunk 5]
```diff
@@ -38,7 +38,18 @@ constexpr uint32_t CORE_APP = 1;
...
+
+#pragma pack(push, 1)
+struct BlePublAdcData {
+	uint16_t status;
+	uint8_t  data[sizeof(ADS131M0x::AdcRawOutput::data)];
+	uint8_t  crc;
+};
+#pragma pack(pop)
+
+constexpr size_t ADC_STREAM_TRIGGER =
```
[the remark this answers is not in the document]
Comment (dl-dl) [reply main/main.cpp:50]: or ADC_BLE_FEED_CHUNK

File: main/ADS131M0x.cpp [hunk 6]
```diff
@@ -1,24 +1,14 @@
 #include "ADS131M0x.h"
-#include "Arduino.h"
+// #include "Arduino.h"
```
Comment (vladkvit) [inline main/ADS131M0x.cpp:2]: Consider removing the comment

File: main/main.cpp (hunk 5 above)
[the remark this answers is not in the document]
Comment (larryk3) [reply main/main.cpp:50]: `ADC_FEED_CHUNK`
```
// To increase throughput, ADC data is grouped and sent as a chunk of multiple readings.
// Send the maximum amount of adc readings that can fit in a BLE payload.
```

[the remark this answers is not in the document]
Comment (dl-dl) [reply main/main.cpp]: Done.

File: main/ADS131M0x.cpp (hunk 6 above)
Comment (dl-dl) [reply main/ADS131M0x.cpp:2]: No.

**Review by larryk3 — APPROVED** [review]

**Pull request merged**